packages/datadog-plugin-aws-sdk/src/services/bedrockruntime.js

```diff
@@ -0,0 +1,236 @@
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.top_p || '',
+      top_k: requestBody.top_k || ''
+    }
+  } else if (provider === STABILITY) {
+    // TODO: request/response formats are different for image-based models. Defer for now
+    return {}
+  }
+  return {}
+}
+
+function extractTextAndResponseReason (response, provider, modelName, shouldSetChoiceIds) {
+  const body = JSON.parse(Buffer.from(response.body).toString('utf8'))
+  const textAndResponseReasons = []
+
+  try {
+    if (provider === AI21) {
+      const completions = body.completions || []
+      completions.forEach(completion => {
+        textAndResponseReasons.push({
+          text: completion.data ? completion.data.text : '',
+          finish_reason: completion.finishReason || '',
+          choice_id: shouldSetChoiceIds ? completion.id : undefined
+        })
+      })
+    } else if (provider === AMAZON && modelName.includes('embed')) {
+      textAndResponseReasons.push({
+        text: body.embedding || [],
+        finish_reason: '',
+        choice_id: undefined
+      })
+    } else if (provider === AMAZON) {
+      (body.results || []).forEach(result => {
+        textAndResponseReasons.push({
+          text: result.outputText || '',
+          finish_reason: result.completionReason || '',
+          choice_id: undefined
+        })
+      })
+    } else if (provider === ANTHROPIC) {
+      textAndResponseReasons.push({
+        text: body.completion || body.content || '',
+        finish_reason: body.stop_reason || '',
+        choice_id: undefined
+      })
+    } else if (provider === COHERE && modelName.includes('embed')) {
+      (body.embeddings || [[]]).forEach(embedding => {
+        textAndResponseReasons.push({
+          text: embedding,
+          finish_reason: '',
+          choice_id: undefined
+        })
+      })
+    } else if (provider === COHERE) {
+      const generations = body.generations || []
+      generations.forEach(generation => {
+        textAndResponseReasons.push({
+          text: generation.text,
+          finish_reason: generation.finish_reason,
+          choice_id: shouldSetChoiceIds ? generation.id : undefined
+        })
+      })
+    } else if (provider === META) {
+      textAndResponseReasons.push({
+        text: body.generation || '',
+        finish_reason: body.stop_reason || '',
+        choice_id: undefined
+      })
+    } else if (provider === MISTRAL) {
+      const generations = body.outputs || []
+      generations.forEach(generation => {
+        textAndResponseReasons.push({
+          text: generation.text,
+          finish_reason: generation.stop_reason
+        })
+      })
+    } else if (provider === STABILITY) {
+      // TODO: request/response formats are different for image-based models. Defer for now
+    }
+  } catch (error) {
+    log.error('Unable to extract text/finish_reason from response body. Defaulting to empty text/finish_reason.')
```

Comment:
i would probably have this as a `log.warn`, i think ideally we reserve a `log.error` for a user mis-configuring the plugin itself (via `tracer.use(...)`), and if it's a problem with the plugin interaction itself we can just `warn` it.